tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -462,6 +469,36 @@ public synchronized void forEach(ElementVisitor reader) throws IOException {
     }
   }
 
+  /**
+   * First reads header to update element counts and file length, then scans each element in the
+   * queue for its length. Will check that none of these lengths are greater than the file size and
+   * that when we reach the last element we are at the position of the last element as recorded in
+   * the header.
+   *
+   * @throws IOException Detailing the corruption identified while scanning.
+   */
+  public synchronized void checkQueueIntegrity() throws IOException {
+    // Read header to update all our counts.
+    readHeaderWithoutFileSideEffects();
+    int position = first.position;
+    int totalSize = HEADER_LENGTH;
+    for (int i = 0; i < elementCount; i++) {
+      if (i == elementCount - 1 && position != last.position) {
+        throw new IOException("Corrupted Element lengths in queue resulted in mis-alignment.");
+      }
+      Element current = readElement(position);
+      totalSize += current.length + Element.HEADER_LENGTH;
+      if (current.length > (fileLength - HEADER_LENGTH)) {
```

Comment:
Should `current.length` be replaced with `totalSize` here?

Reply:
No i meant to check if any single element's length is out of portion with the file size. the totalSize is checked at the end.